docs/rfcs/0003-rfc-controller_refactor.md

```diff
@@ -0,0 +1,112 @@
+- Status: In review
+- Date: 2023-06-20
+- Author: [@gela](https://github.com/gela)
+- RFC PR: https://github.com/mozilla-mobile/mozilla-vpn-client/pull/7269
+- Implementation GitHub issue: TBD.
+
+
+## Summary
+Introduce a new `ConnectivityManager` component which sits between the frontend and controller to handle all connectivity related logic and keep the controller code lean and minimal.
+
+## Motivation
+The current controller code is complicated, lengthy and bug-prone. Over time we have continued to directly add more logic and checks for various scenarios to the controller code. 
+
+Because our current approach is not scalable, we’ve been facing numerous issues in the connectivity space, especially when it comes down to identifying exactly what is causing the issue. This means the behavior of our client is extremely inconsistent both in different platforms but also within the same environment. For example if a user loses network connectivity while the VPN is activated, sometimes they enter no signal, and other times they encounter the “server location unavailable” modal. This inconsistency is a result of racing checks and error reporting. The goal of this refactor is to combine all of these scenarios under one component and to get rid of the Connection Health entirely to resolve this issue.
+
+
+## Proposed Solution
+Move all connectivity related code out of the controller and into its own separate component which is referred to as the Connectivity Manager in this document. The controller object will continue to remain but its purpose will be drastically simplified to a few basic functionalities including `activate()` and `deactivate()`. The connectivity manager will primarily act as a liaison between the controller and the UI. 
+
+```mermaid
+graph TD
+    Controller --> ConnectivityManager
+    ConnectivityManager --> Frontend
+    Frontend --> Controller
```

Comment:
Above we say the connectivity manager `sits between the frontend and controller` but here it looks like there is still some communication from the frontend to the controller. Is this intentional?